crates/bevy_ecs/src/schedule/condition.rs

```diff
@@ -141,4 +143,30 @@ pub mod common_conditions {
     {
         condition.pipe(|In(val): In<bool>| !val)
     }
+
+    /// Generates a [`Condition`](super::Condition)-satisfying closure that returns `true`
+    /// if there are any entities with the added given component type.
+    ///
+    /// Run conditions are evaluated on the main thread, blocking any other systems from running.
+    /// This run condition is relatively expensive, as it iterates over every entity with this component.
+    /// As a result, you likely only want to use this run condition when the number of entitities with the component `T` is small.
+    pub fn any_component_added<T: Component>() -> impl FnMut(Query<(), Added<T>>) -> bool {
+        move |query: Query<(), Added<T>>| !query.is_empty()
```

Comment:
These potentially do a full world scan in the multithreaded executor, as the only early return is if no archetypes match the query. This is relatively easy performance footgun, same with the other condition below.

Reply:
Yep, this is what I mentioned in the description. I thought it's fine if documented. But I understand the controversy. This is why it wasn't included in https://github.com/bevyengine/bevy/pull/7579.